cray_product_catalog/catalog_delete.py

```diff
@@ -61,19 +61,167 @@
+        self.__max_reties_for_prod_cm = __max_reties_for_prod_cm
+
+    @property
+    def key(self):
+        return self.__key
+
+    @key.setter
+    def key(self, __key):
+        self.__key = __key
+
+    @property
+    def main_cm_fields(self):
+        return self.__main_cm_fields
+
+    @main_cm_fields.setter
+    def main_cm_fields(self, __main_cm_fields):
+        self.__main_cm_fields = __main_cm_fields
+
+    @property
+    def product_cm_fields(self):
+        return self.__product_cm_fields
+
+    @product_cm_fields.setter
+    def product_cm_fields(self, __product_cm_fields):
+        self.__product_cm_fields = __product_cm_fields
+
+    # private methods
+    def __key_belongs_to_main_cm_fields(self):
+        return self.__key in self.__main_cm_fields
+
+    def __key_belongs_to_prod_cm_fields(self):
+        return self.__key in self.__product_cm_fields
+
+    def __modify_main_cm(self):
+        LOGGER.info("Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
+                    self.main_cm, self.cm_namespace, self.product_name, self.product_version, self.key)
+        modify_config_map(self.__main_cm, self.__cm_namespace, self.__product_name, self.__product_version,
+                          self.__key, self.__max_reties_for_main_cm,)
+
+    def __modify_product_cm(self):
+        LOGGER.info("Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
+                    self.product_cm, self.cm_namespace, self.product_name, self.product_version, self.key)
+        modify_config_map(self.__product_cm, self.__cm_namespace, self.__product_name, self.__product_version,
+                          self.__key, self.__max_reties_for_prod_cm,)
+
+    # public method
+    def modify_config_map(self):
+        """
+        Method to initiate modification of ConfigMaps.
+        Before executing this method make sure to set these properties of the class:
+        *    main_cm # name of main ConfigMap
+        *    product_cm # name of product specific CofigMap
+        *    cm_namespace  # Namespace containing all config map
+        *    product_name  # Product name
+        *    product_version  # Product version
+        *    max_reties_for_main_cm  # Max failure retries for main ConfigMap
+        *    max_reties_for_prod_cm  # Max failure retries for product ConfigMap
+        *    key  # Key to delete, if you want to execute complete product or a particular version, ignore it
+        *    main_cm_fields  # Fields present in main ConfigMap
+        *    product_cm_fields  # Fields present in product specific ConfigMap
+        """
+        if self.__key:
+            if self.__key_belongs_to_main_cm_fields():
+                self.__modify_main_cm()
+
+            elif self.__key_belongs_to_prod_cm_fields():
+                self.__modify_product_cm()
+
+            else:
+                LOGGER.error("Invalid KEY=%s is input so exiting...", self.key)
```

Comment:
```suggestion
                LOGGER.warning("Key=%s, is not present in Main/Product ConfigMap", self.key)
```

Reply:
Agreed, it should be warning and modified the logger string.